ingestion/src/metadata/ingestion/lineage/sql_lineage.py

```diff
@@ -315,17 +315,25 @@ def _create_lineage_by_table_name(
             table_name=to_table,
         )
 
-        for from_entity in from_table_entities or []:
-            for to_entity in to_table_entities or []:
-                yield from _build_table_lineage(
-                    to_entity=to_entity,
-                    from_entity=from_entity,
-                    to_table_raw_name=to_table,
-                    from_table_raw_name=from_table,
-                    query=query,
-                    column_lineage_map=column_lineage_map,
-                    lineage_source=lineage_source,
-                )
+        for table_name, entity in (
+            (from_table, from_table_entities),
+            (to_table, to_table_entities),
+        ):
+            if entity is None:
+                raise RuntimeError(f"Table entity not found: [{table_name}]")
+
```

Comment:
I'm not sure if I like idea of raising error here, consider this scenario

```
insert into abc as (select id from xyz union select id from pqr)
```

if i do not fine table `pqr` here, lineage between abc and xyz will also not get ingested, how about just logging that the table was not found?

Reply:
good point. Ill add this as a test case and look for a solution

Reply:
@ulixius9 looked into this and added a test case [here](https://github.com/open-metadata/OpenMetadata/pull/16756/files#diff-4b742c40aa666c03af4c10d2111fdd67f00477f040a2d8f304b9e42e6e5dcbaaR5). If a query has multiple tables, it will register an error for failed ones and add the ones which exist.